File: src/main/java/ba/aljovic/amer/movierecommendation/configuration/DecisionTreeConfig.java
package ba.aljovic.amer.movierecommendation.configuration;

import ba.aljovic.amer.movierecommendation.application.algorithm.Recommendation;
import ba.aljovic.amer.movierecommendation.application.algorithm.RecommendationAlgorithm;
import ba.aljovic.amer.movierecommendation.application.algorithm.DecisionTreeAlgorithm;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

@Configuration
@Profile ("algorithm-decision-tree")
public class DecisionTreeConfig
{
    @Bean
    public Recommendation decisionTreeRecommendation()
    {
        return new Recommendation.RecommendationBuilder(decisionTreeAlgorithm()).build();
    }

    @Bean
    public RecommendationAlgorithm decisionTreeAlgorithm()
    {
        return new DecisionTreeAlgorithm(100);
    }
}

File: src/main/java/ba/aljovic/amer/movierecommendation/configuration/DecisionTreeConfig.scala
package ba.aljovic.amer.movierecommendation.configuration

import ba.aljovic.amer.movierecommendation.application.algorithm.{Recommendation, RecommendationAlgorithm, DecisionTreeAlgorithm}
import org.springframework.context.annotation.{Bean, Configuration, Profile}

@Configuration
@Profile(Array("algorithm-decision-tree"))
class DecisionTreeConfig
{
  @Bean
  def decisionTreeRecommendation: Recommendation = {
    Recommendation(decisionTreeAlgorithm)
  }

  @Bean
  def decisionTreeAlgorithm: RecommendationAlgorithm = new DecisionTreeAlgorithm(5, 9, 10, 3)
}
